core/src/main/kotlin/io/customer/sdk/core/di/AndroidSDKComponent.kt

```diff
@@ -20,8 +21,10 @@ class AndroidSDKComponent(
     val context: Context,
     val client: Client
 ) : DiGraph() {
+    val application: Application
+        get() = newInstance<Application> { context.applicationContext as Application }
```

Comment:
Ideally, application context should always be an instance of Application. So I don't think we need to handle any nullable scenario here, but open to ideas.